runtime/mux.go

```diff
@@ -204,6 +206,50 @@ func WithDisablePathLengthFallback() ServeMuxOption {
 	}
 }
 
+// WithHealthzEndpoint returns a ServeMuxOption that will add a /healthz endpoint to the created ServeMux.
+// When called the handler will forward the request to the upstream grpc service health check (defined in the
+// gRPC Health Checking Protocol).
+// See here https://grpc-ecosystem.github.io/grpc-gateway/docs/operations/health_check/ for more information on how
+// to setup the protocol in the grpc server.
+// If you define a service as query parameter, this will also be forwarded as service in the HealthCheckRequest.
+func WithHealthzEndpoint(conn grpc.ClientConnInterface) ServeMuxOption {
+	healthCheckClient := grpc_health_v1.NewHealthClient(conn)
```

Comment:
I know I've already reviewed this, but now that I'm looking at it, wouldn't it make more sense for this parameter to be an instantiation of the health client? That way it'll be easier to test and the user can decide how to dial the client.
```suggestion
func WithHealthzEndpoint(healthCheckClient grpc_health_v1.HealthClient) ServeMuxOption {
```

Reply:
That was a really nice hint, feels much cleaner now. I was not sure if you are using any mocking framework and couldn't find one, so I kept the fake approach.